### PR TITLE
IPC balance changes and power failure

### PR DESCRIPTION
**Pull request by KittyNoodle**
## What Does This PR Do
Makes IPCs cold immune and take slightly less brute damage(they still take their massive burn damage however).
IPCs now take 130% brute and 150% burn.
IPCs now take brain damage from extreme radiation. 
IPCs now have a hypoglycemia-like condition when low on power.
Inspiration taken from beestation's IPC damage multipliers 

## Why It's Good For The Game

IPCs in their current state are extremely weak. This will give them a slight buff while still preserving their large damage amount with burn damage. Hopefully meaning that IPCs will withstand more than one toolbox hit, while still dying to welders and other burn damage sources.
IPCs also are the only race that does not have major penalties from starvation, this pr changes that.

## Progress

- [x]  Radiation Damage

- [x] Power Failure

## Changelog
:cl:
add: IPCs now take brain damage when exposed to high level rads.
add: IPCs now actually require power to live.
tweak: IPCs now take less brute damage and are cold immune.
/:cl:

### CODE REVIEW

**Comment by Calyxman** [comment]
Awesome.

**Comment by Coolrune206** [comment]
Is that a change from 50% extra brute damage to 20%?

Might be worth recording that in the changelog/PR proper so people can see it easily.

Also, I don't really think IPCs need a buff overall - their current fragility to some sources of damage and complete immunity to others gives them a unique identity which I'm quite fond of.

File: code/modules/mob/living/carbon/human/species/machine.dm [hunk 1]
```diff
@@ -14,15 +14,15 @@
 	skinned_type = /obj/item/stack/sheet/metal // Let's grind up IPCs for station resources!
 
 	eyes = "blank_eyes"
-	brute_mod = 2.28 // 100% * 2.28 * 0.66 (robolimbs) ~= 150%
-	burn_mod = 2.28  // So they take 50% extra damage from brute/burn overall
+	brute_mod = 1.97 // 100% * 1.82 * 0.66 (robolimbs) ~= 130%
```
Comment (Edan52) [inline code/modules/mob/living/carbon/human/species/machine.dm:17]: Comment needs updating, also it's a good idea to actually specify the number changes in the PR description

Comment (KittyNoodle) [reply code/modules/mob/living/carbon/human/species/machine.dm]: done

**Comment by Calyxman** [comment]
Everything's alright about this (maybe IPCs could take power from live wires or generators as well?) but not sure how I feel about the IPC brain damage though, if it's really high maybe some sort of IPC radiation hardening procedure or surgery to lessen/negate it?

Trying to conserve battery and plugging into emergency generators just to survive.. sounds interesting

**Comment by meow20** [comment]
> Everything's alright about this (maybe IPCs could take power from live wires or generators as well?) but not sure how I feel about the IPC brain damage though, if it's really high maybe some sort of IPC radiation hardening procedure or surgery to lessen/negate it?


There is ipc eqivalent of mannitol it's just never used because there is no reason to so im glad about that part of the PR, radiation is also so rare that doing a surgery to negate it makes not much sense.

**Comment by Calyxman** [comment]
> > Everything's alright about this (maybe IPCs could take power from live wires or generators as well?) but not sure how I feel about the IPC brain damage though, if it's really high maybe some sort of IPC radiation hardening procedure or surgery to lessen/negate it?
> 
> There is ipc eqivalent of mannitol it's just never used because there is no reason to so im glad about that part of the PR, radiation is also so rare that doing a surgery to negate it makes not much sense.

Rads aren't really too rare, take engine operation for example or radiation contamination

File: code/modules/mob/living/carbon/human/species/machine.dm [hunk 2]
```diff
@@ -101,6 +101,12 @@
 	H.med_hud_set_health()
 	H.med_hud_set_status()
 
+/datum/species/machine/spec_rad_act(amount, mob/living/carbon/human/H)
+	if(amount >= RAD_MINIMUM_CONTAMINATION)
+		if(prob(20)
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/human/species/machine.dm:106]:
```suggestion
		if(prob(20))
```
missing a bracket here.

**Comment by KittyNoodle** [comment]
The way I set up ipc radiation damage is quite different to how other species handle rad damage, they do not take doses and instead just take damage when directly irradiated.
As for making them able to sap power directly from generators that sounds like a nightmare to code, I am however working on more ways to give ipcs power, but that will be in a different pr.

**Comment by MRgaty** [comment]
If IPCs are cold inmune that would mean they would only need a fire suit and a helmet to be space proof. I dislike such idea

**Comment by AffectedArc07** [comment]
PR has 3 objections from within leadership, with no one in favour at all.

Closing.

**Pull request closed without merging**